core/src/main/java/org/apache/calcite/sql/validate/implicit/TypeCoercionImpl.java

```diff
@@ -535,7 +535,7 @@ public int size() {
           coerced = rowTypeCoercion(scope1, node2, i, desired) || coerced;
         }
```

Comment:
Add this `coerced = coerced || listCoerced;` code in line 530 is enough, can we try to give a test case here ?

Reply:
I'll give it a try.

Reply:
> Add this `coerced = coerced || listCoerced;` code in line 530 is enough, can we try to give a test case here ?

Hi Danny, I've updated my PR and add a test, would you take a look?